hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/proto/FederationProtocol.proto

```diff
@@ -311,17 +311,4 @@ message GetDisabledNameservicesRequestProto {
 
 message GetDisabledNameservicesResponseProto {
   repeated string nameServiceIds = 1;
-}
-
-/////////////////////////////////////////////////
-// Alignment state for namespaces.
-/////////////////////////////////////////////////
-
-/**
- * Clients should receive this message in RPC responses and forward it
- * in RPC requests without interpreting it. It should be encoded
- * as an obscure byte array when being sent to clients.
- */
-message RouterFederatedStateProto {
-  map<string, int64> namespaceStateIds = 1; // Last seen state IDs for multiple namespaces.
```

Comment:
Can we change the protocol so freely?

Reply:
Sir, do you have some good suggestions? ClientGSIContext needs RouterFederatedStateProto, so moved it to hdfs.proto

Reply:
@ayushtkn do you think is fine moving this around?

Reply:
Technically by compat such changes aren't allowed

But HADOOP-18345 hasn't been released yet, so technically if you commit this to 3.3.5 as well, so this change never existed in a released version, so can be changed. 

3.3.5 I suppose will be out for vote in a week or two, so, if this code lands up having any issues, you can't come back though. :-)

Reply:
Thanks @ayushtkn, then there's a window here.
@simbadzina @omalley can you help review as this touches code you changed in HADOOP-18345?

Reply:
I'm in support of moving RouterFederatedStateProto to hdfs.proto. From a modularity point of view we'll be removing a dependence of hadoop-hdfs-client on hadoop-hdfs-rbf, which makes the code cleaner.
In HDFS-18345 I thought that clients shouldn't have visibility into this message, but I now see that it is necessary for clients to interact with the message as this pull request shows. So moving it to hdfs.proto is a good change.